syft/frameworks/torch/tensors/interpreters/numpy.py

```diff
@@ -0,0 +1,73 @@
+from syft.generic.frameworks.hook.hook_args import (
+    register_type_rule,
+    register_forward_func,
+    register_backward_func,
+    one,
+)
+
+from syft.generic.tensor import AbstractTensor
+
+
+class NumpyTensor(AbstractTensor):
+    """NumpyTensor is a tensor which seeks to wrap the Numpy API with the PyTorch tensor API.
+    This is useful because Numpy can offer a wide range of existing functionality ranging from
+    large precision, custom scalar types, and polynomial arithmetic.
+    """
+
+    def __init__(
+        self,
+        numpy_tensor=None,
+        owner=None,
+        id=None,
+        tags=None,
+        description=None,
+        verbose=False,
+    ):
+        """Initializes a NumpyTensor.
+
+        Args:
+            numpy_tensor (np.array): The numpy array which this tensor should wrap.
+            owner (BaseWorker): An optional BaseWorker object to specify the worker on which
+                the tensor is located.
+            id (str or int): An optional string or integer id of the LargePrecisionTensor.
+            tags (list): list of tags for searching.
+            description (str): a description of this tensor.
+        """
+        super().__init__(id=id, owner=owner, tags=tags, description=description)
+        self.verbose = verbose
+        self.child = numpy_tensor
+
+    def __add__(self, other):
+        return NumpyTensor(numpy_tensor=self.child + other.child)
```

Comment:
Hey if you use `overloaded` you can make that nice stuff:
```python
@overloaded.method
def add(self, _self, other):
    return _self + other
```
And you spare yourself extra efforts :)

Reply:
@iamtrask extra bonus: register your tensor to the hook.py and ask to overload the methods: it will support by default all those which are equivalent in torch and numpy an you will just have to rewrtie those which changed!

Reply:
I'm considering creating a new tensor type whose job is just to be hooked. That way we can extend it like AbstractTensor

Reply:
Done